### DISCUSSION TITLE
Recommended caching headers to prevent caching

### DISCUSSION THREAD

**Discussion by rickstaa** [vercel/vercel#8741] [Help] [answered]
In one of [our repositories](https://github.com/anuraghazra/github-readme-stats/blob/98f9045f987112bb3d4f1a0160f3e0d93ef5bd52/api/index.js#L91-L93), we display an error SVG when [GitHub's GraphQL rate limit](https://docs.github.com/en/graphql/overview/resource-limitations) is being hit. To prevent this error SVG from being cached, I added the `Cache-Control: no-store` header. It looks like chrome does respect this header and never serves the error SVG from the cache. FireFox, on the other hand, sometimes seems to ignore this header and serve the cached error SVG (see video below). The whole discussion of this problem can be found [here](https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1282585613).

https://user-images.githubusercontent.com/17570430/196475753-feb94563-c547-4a72-9850-dd935bb32082.mp4

After reading [this StackOverflow answer](https://stackoverflow.com/questions/49547/how-do-we-control-web-page-caching-across-all-browsers/2068407#2068407), I changed the header to `Cache-Control: no-cache, no-store, must-revalidate (see https://github.com/anuraghazra/github-readme-stats/blob/98f9045f987112bb3d4f1a0160f3e0d93ef5bd52/api/index.js#L91-L94) but I was wondering what header you recommend for preventing the error SVG from being cached.

### EDIT

I think the problem was caused by the GitHub CDN not respecting the `no-store` header (see https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#troubleshooting-issues-with-camo and https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1282702795). I, however, am still curious about what headers you would recommend :nerd:.

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @rickstaa. If your solution is working for you then I think you have already found the answer! 😁 

For edge network and serverless function caching, there are some tips for how to cache which could be used as reference to _avoid_ caching: 
- [Edge Network Cacheable Responses](https://vercel.com/docs/concepts/edge-network/caching#cacheable-responses)
- [Caching responses from Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching)

**Reply by rickstaa** [reply]
@amyegan Thanks for your answer. I can not verify if it works since the bug is currently not occurring. Judging from the [GitHub's](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#troubleshooting-issues-with-camo) and [your documentation](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching) however, I'm confident that it works now or at least is not a Vercel problem. 👍🏻
